File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        left, right, max_val = 0, 0, 0
        while right < len(prices):
            if right > 0 and right != left and prices[right] < prices[left]:
                left = right
            else:
                diff = prices[right] - prices[left]
                max_val = max(diff, max_val)
                print(left, right, diff, max_val)
            right+=1
            print('pointers: ', left, right)
        return max_val





        

File: Data Structures & Algorithms/max-water-container/submission-1.js
class Solution {
    /**
     * @param {number[]} heights
     * @return {number}
     */
    maxArea(heights) {
        let left = 0, right = heights.length - 1, max = 0;
        while(left < right){
            const currMax = Math.min(heights[left], heights[right]) * (right - left);
            max = Math.max(currMax, max);
            if(heights[left] <= heights[right]) left++; 
            else right--; 
        }

        return max; 
    }
}
